File: backend/src/main/java/com/bigdata/etl/model/entity/Datasource.java
package com.bigdata.etl.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("etl_datasource")
public class Datasource {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String type;       // MYSQL, POSTGRESQL, ORACLE, SQLSERVER
    private String host;
    private Integer port;
    private String database;
    private String username;
    private String password;
    private String status;     // ACTIVE, INACTIVE
    private String remark;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/datasource.ts
import request, { type ApiResponse } from './index'

export interface Datasource {
  id?: number
  name: string
  type: string // MYSQL / POSTGRESQL / ORACLE / SQLSERVER
  host: string
  port: number
  database: string
  username: string
  password?: string // 后端 WRITE_ONLY，列表不返回
  status?: string // ACTIVE / INACTIVE
  remark?: string
}

export const datasourceApi = {
  list: () => request.get<unknown, ApiResponse<Datasource[]>>('/datasources'),
  create: (data: Datasource) => request.post<unknown, ApiResponse<Datasource>>('/datasources', data),
  update: (id: number, data: Datasource) =>
    request.put<unknown, ApiResponse<void>>(`/datasources/${id}`, data),
  delete: (id: number) => request.delete<unknown, ApiResponse<void>>(`/datasources/${id}`),
  // 后端仅支持按已保存数据源 ID 测试连接：POST /datasources/{id}/test
  test: (id: number) => request.post<unknown, ApiResponse<string>>(`/datasources/${id}/test`)
}
